src/Compilers/CSharp/Test/Emit/Attributes/AttributeTests_WellKnownAttributes.cs

```diff
@@ -10321,6 +10321,72 @@ partial class C
             Assert.False(comp.HasLocalsInit("C.M"));
         }
 
+        [Fact]
+        public void SkipLocalsInitAttributeOnExtendedPartialMethod_01()
+        {
+            var source = @"
+namespace System.Runtime.CompilerServices
+{
+    class SkipLocalsInitAttribute : System.Attribute
+    {
+    }
+}
+
+partial class C
+{
+    public partial int M()
+    {
+        int x = 1;
+        x = x + x + x;
+        return x;
+    }
+}
+
+partial class C
+{
+    [System.Runtime.CompilerServices.SkipLocalsInitAttribute]
+    public partial int M();
+}
+";
+
+            var comp = CompileAndVerify(source, options: TestOptions.UnsafeReleaseDll, parseOptions: TestOptions.RegularWithExtendedPartialMethods, verify: Verification.Fails);
```

Comment:
>verify: Verification.Fails [](start = 148, length = 26)

Why does verification fail?

Reply:
I believe the verifier is unhappy that locals are not initialized. This is similar to other SkipLocalsInit tests.

Reply:
Are we skipping init? If so, why is `HasLocalsInit("C.M")` returning `false`?

---
In reply to: [426860257](https://github.com/dotnet/roslyn/pull/44303#discussion_r426860257) [](ancestors = 426860257)

Reply:
Yes. `HasLocalsInit` is returning `false` because we are skipping locals init. Normally it returns true in most cases, except when the method has no locals or when the SkipLocalsInitAttribute has been applied.